gulpfile.js

```diff
@@ -547,24 +551,38 @@ function deployCesium(bucketName, uploadDirectory, cacheControl, done) {
 }
 
 function getMimeType(filename) {
-    var ext = path.extname(filename);
-    if (ext === '.bin' || ext === '.terrain') {
-        return {type : 'application/octet-stream', compress : true, isCompressed : false};
-    } else if (ext === '.md' || ext === '.glsl') {
-        return {type : 'text/plain', compress : true, isCompressed : false};
-    } else if (ext === '.czml' || ext === '.geojson' || ext === '.json') {
-        return {type : 'application/json', compress : true, isCompressed : false};
-    } else if (ext === '.js') {
-        return {type : 'application/javascript', compress : true, isCompressed : false};
-    } else if (ext === '.svg') {
-        return {type : 'image/svg+xml', compress : true, isCompressed : false};
-    } else if (ext === '.woff') {
-        return {type : 'application/font-woff', compress : false, isCompressed : false};
+    var mimeType = mime.getType(filename);
+    if (mimeType) {
+        //Compress everything except binary images and video
+        var compress = !/^(image|video)\//i.test(mimeType);
+        if (mimeType === 'image/svg+xml') {
+            compress = true;
+        }
+        return { type: mimeType, compress: compress };
     }
 
-    var mimeType = mime.getType(filename);
-    var compress = compressible(mimeType);
-    return {type : mimeType, compress : compress, isCompressed : false};
+    //Non-standard mime types not handled by mime
+    if (/\.(glsl|LICENSE|config|state)$/i.test(filename)) {
+        return { type: 'text/plain', compress: true };
+    } else if (/\.(czml|topojson)$/i.test(filename)) {
+        return { type: 'application/json', compress: true };
+    } else if (/\.crn$/i.test(filename)) {
+        return { type: 'application/octet-stream', compress: false };
+    }
+
+    // Handle dotfiles, such as .jshintrc
```

Comment:
More of a general question, should we be deploying _any_ dotfiles besides .jshintrc?

Reply:
That is a good question, right now it's certainly intentional in the existing script but I'm not sure it's needed.  Feel free to write up an issue and we can re-evaluate in the future.